### PR TITLE
SHIP-0004: Build Environment Variables

### PR DESCRIPTION
**Pull request by coreydaley**
Adding documentation for SHIP-0004: Build Environment Variables

### CODE REVIEW

**Comment by adambkaplan** [comment]
@coreydaley for API docs we write the doc in the repository with the code (shipwright-io/build in this case), and then sync the documentation to the website at release time.

I did this manually the last time - we have a TODO to automate this in #29.

File: content/en/docs/api/build.md [hunk 1]
```diff
@@ -144,6 +148,24 @@ spec:
...
+  name: buildah-golang-build
+spec:
+  source:
+    url: https://github.com/shipwright-io/sample-go
+    contextDir: docker-build
+  env:
+    - name: EXAMPLE_VAR_1
+      value: "example-value-1"
+    - name: EXAMPLE_VAR_2
+      value: "example-value-2"
```
Comment (gabemontero) [inline content/en/docs/api/build.md:166]: aside from the where you initially specify content comment from @adambkaplan - my comment in the shipwright-io/build PR about the type of this field would come into play

if this field is going to be k8s corev1.EnvVar, then we need more examples to show getting content from configmaps, secrets, etc.

or our doc cites those are not supported

if, however, we go with a a simpler type that is a subset of corev1.EnvVar, then these examples seem sufficient

**Comment by coreydaley** [comment]
/close
Added documentation to https://github.com/shipwright-io/build/pull/817

File: content/en/docs/api/build.md (hunk 1 above)
Comment (coreydaley) [reply content/en/docs/api/build.md:166]: Moved documentation to https://github.com/shipwright-io/build/pull/817

**Pull request closed without merging**